### PR TITLE
Add wrapping for SLICImageFilter from SimpleITKFilters module

### PR DESCRIPTION
**Pull request by blowekamp**
This filter is currently in the SimpleITKFilters remote module, it is
also in the ITKv5 main repository.

### CODE REVIEW

File: Code/BasicFilters/json/SLICImageFilter.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      "detaileddescriptionSet" : "The requested size of a superpixel used to form a regular grid for initialization and limits the search space for pixels. The size may be set anisotropically to provide a directional bias. This may be set to reflect spacing of this image.",
+      "briefdescriptionGet" : "The expected superpixel size and shape.",
+      "detaileddescriptionGet" : "The requested size of a superpixel used to form a regular grid for initialization and limits the search space for pixels. The size may be set anisotropically to provide a directional bias. This may be set to reflect spacing of this image."
+    },
+    {
+      "name" : "SpatialProximityWeight",
+      "type" : "double",
+      "default" : "10.0",
+      "briefdescriptionSet" : "The spatial weight for the distance function.",
+      "detaileddescriptionSet" : "Increasing this value makes the superpixel shape more regular, but more varied in image values. The range of the pixel values and image dimension can effect the appropriate value.",
```
Comment (zivy) [inline Code/BasicFilters/json/SLICImageFilter.json]: Possibly say that the default value is appropriate for 2D images in the CIE-Lab color space?

File: Code/BasicFilters/json/SLICImageFilter.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
... a set of labeled regions or super-pixels. Super-pixels follow natural image boundaries, are compact, and are nearly uniform regions which can be used as a larger primitive for more efficient computation. The SLIC algorithm can be viewed as a spatially constrained iterative k-means method.\n\nThe original algorithm was designed to cluster on the joint domain of the images index space and it's CIELAB color space. This implementation works with images of arbitrary dimension as well as scalar, single channel, images and most multi-component image types including ITK's arbitrary length VectorImage .\n\nThe distance between a pixel and a cluster is the sum of squares of the difference between their joint range and domains ( index and value ). The computation is done in index space with scales provided by the SpatialProximityWeight parameters.\n\nThe output is a label image with each label representing a superpixel cluster. Every pixel in the output is labeled, and the starting label id is zero.\n\nThis code was contributed in the Insight Journal paper: \"Scalable Simple Linear Iterative Clustering (SSLIC) Using a\nGeneric and Parallel Approach\" by Lowekamp B. C., Chen D. T., Yaniv Z.",
```
Comment (zivy) [inline Code/BasicFilters/json/SLICImageFilter.json]: Reference doesn't include Terry's name at the end (it is included in the ITK doxygen, though the link to the paper is missing there, we should probably add it).

**Review by zivy — APPROVED** [review]
LGTM, only two minor documentation comments.

**Comment by blowekamp** [comment]
The filter comes from https://github.com/SimpleITK/ITKSimpleITKFilters, these changes would need to be applied there.

**Pull request merged**